diff --git a/arch/mips/jz4730/pm.c b/arch/mips/jz4730/pm.c
index 9ea71c4..2e0a295 100644
--- a/arch/mips/jz4730/pm.c
+++ b/arch/mips/jz4730/pm.c
@@ -14,7 +14,6 @@
 
 #include <linux/init.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/sched.h>
 #include <linux/proc_fs.h>
 #include <linux/sysctl.h>
diff --git a/arch/mips/jz4740/pm.c b/arch/mips/jz4740/pm.c
index 8ad2a57..6f96bdc 100644
--- a/arch/mips/jz4740/pm.c
+++ b/arch/mips/jz4740/pm.c
@@ -23,7 +23,6 @@
 
 #include <linux/init.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/sched.h>
 #include <linux/interrupt.h>
 #include <linux/proc_fs.h>
diff --git a/arch/mips/jz4750/pm.c b/arch/mips/jz4750/pm.c
index a1e935c..1a768f2 100644
--- a/arch/mips/jz4750/pm.c
+++ b/arch/mips/jz4750/pm.c
@@ -23,7 +23,6 @@
 
 #include <linux/init.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/sched.h>
 #include <linux/interrupt.h>
 #include <linux/proc_fs.h>
diff --git a/arch/mips/jz4750d/pm.c b/arch/mips/jz4750d/pm.c
index b8cb4a8..0c2899e 100644
--- a/arch/mips/jz4750d/pm.c
+++ b/arch/mips/jz4750d/pm.c
@@ -23,7 +23,6 @@
 
 #include <linux/init.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/sched.h>
 #include <linux/interrupt.h>
 #include <linux/proc_fs.h>
diff --git a/drivers/char/jzchar/ak4182.c b/drivers/char/jzchar/ak4182.c
index f4a970f..f2ec9a8 100644
--- a/drivers/char/jzchar/ak4182.c
+++ b/drivers/char/jzchar/ak4182.c
@@ -19,7 +19,6 @@
 #include <linux/poll.h>
 #include <linux/string.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 
 #include <asm/uaccess.h>
 #include <asm/irq.h>
diff --git a/drivers/char/jzchar/poweroff.c b/drivers/char/jzchar/poweroff.c
index 7127438..4970b68 100644
--- a/drivers/char/jzchar/poweroff.c
+++ b/drivers/char/jzchar/poweroff.c
@@ -24,7 +24,6 @@
 #include <linux/fs.h>
 #include <linux/spinlock.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 
 #include <asm/io.h>
 #include <asm/uaccess.h>
diff --git a/drivers/mmc/host/jz4750_mmc.c b/drivers/mmc/host/jz4750_mmc.c
index c3bae83..a19cd15 100644
--- a/drivers/mmc/host/jz4750_mmc.c
+++ b/drivers/mmc/host/jz4750_mmc.c
@@ -23,7 +23,6 @@
 #include <linux/mm.h>
 #include <linux/signal.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/scatterlist.h>
 
 #include <asm/io.h>
diff --git a/drivers/mmc/host/jz_mmc.c b/drivers/mmc/host/jz_mmc.c
index 67781da..971dbe0 100644
--- a/drivers/mmc/host/jz_mmc.c
+++ b/drivers/mmc/host/jz_mmc.c
@@ -23,7 +23,6 @@
 #include <linux/mm.h>
 #include <linux/signal.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/scatterlist.h>
 
 #include <asm/io.h>
@@ -755,8 +754,11 @@ static const struct mmc_host_ops jz_mmc_ops = {
 	.get_ro = jz_mmc_get_ro,
 	.set_ios = jz_mmc_set_ios,
 };
+#if 0
+/* will fix this later add by Xiangfu Liu*/
 static int jz_mmc_pm_callback(struct pm_dev *pm_dev,
 			      pm_request_t req, void *data);
+#endif
 
 static int jz_mmc_probe(struct platform_device *pdev)
 {
@@ -817,7 +819,7 @@ static int jz_mmc_probe(struct platform_device *pdev)
 	mmc->ocr_avail = host->pdata ?
 		host->pdata->ocr_mask : MMC_VDD_32_33 | MMC_VDD_33_34;
 	host->mmc->caps =
-		MMC_CAP_4_BIT_DATA | MMC_CAP_MULTIWRITE | MMC_CAP_SD_HIGHSPEED
+		MMC_CAP_4_BIT_DATA | MMC_CAP_SD_HIGHSPEED
 		| MMC_CAP_MMC_HIGHSPEED;
 	/*
 	 *MMC_CAP_4_BIT_DATA    (1 << 0)    The host can do 4 bit transfers
@@ -874,9 +876,9 @@ static int jz_mmc_probe(struct platform_device *pdev)
 	mmc_add_host(mmc);
 #ifdef CONFIG_PM
 	/* Register MMC slot as as power-managed device */
-	host->pmdev = pm_register(PM_UNKNOWN_DEV, PM_SYS_UNKNOWN, jz_mmc_pm_callback);
-	if (host->pmdev)
-		host->pmdev->data = pdev;
+	/* host->pmdev = pm_register(PM_UNKNOWN_DEV, PM_SYS_UNKNOWN, jz_mmc_pm_callback); */
+	/* if (host->pmdev) */
+	/* 	host->pmdev->data = pdev; */
 #endif
 	printk("JZ SD/MMC card driver registered\n");
 
@@ -975,16 +977,18 @@ static int jz_mmc_resume(struct platform_device *dev)
 
 	return ret;
 }
+#if 0
+/* will fix this later add by Xiangfu Liu*/
 static int jz_mmc_pm_callback(struct pm_dev *pm_dev,
                                pm_request_t req, void *data)
 {
 	struct platform_device *pdev = (struct platform_device *)pm_dev->data;
 
 	switch(req) {
-	case PM_RESUME:
+	case PM_EVENT_RESUME:
 		jz_mmc_resume(pdev);
 		break;
-	case PM_SUSPEND:
+	case PM_EVENT_SUSPEND:
 	/* state has no use */
 		jz_mmc_suspend(pdev, state);
 		break;
@@ -994,6 +998,8 @@ static int jz_mmc_pm_callback(struct pm_dev *pm_dev,
 	}
 	return 0;
 }
+#endif 
+
 #else
 #define jz_mmc_suspend	NULL
 #define jz_mmc_resume	NULL
diff --git a/drivers/video/jz4740_slcd.c b/drivers/video/jz4740_slcd.c
index 0d893a4..450e233 100644
--- a/drivers/video/jz4740_slcd.c
+++ b/drivers/video/jz4740_slcd.c
@@ -22,7 +22,6 @@
 #include <linux/dma-mapping.h>
 #include <linux/platform_device.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 
 #include <asm/irq.h>
 #include <asm/pgtable.h>
diff --git a/drivers/video/jz4750_lcd.c b/drivers/video/jz4750_lcd.c
index 5350a70..95da4cd 100644
--- a/drivers/video/jz4750_lcd.c
+++ b/drivers/video/jz4750_lcd.c
@@ -35,7 +35,6 @@
 #include <linux/dma-mapping.h>
 #include <linux/platform_device.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 
 #include <asm/irq.h>
 #include <asm/pgtable.h>
diff --git a/drivers/video/jzlcd.c b/drivers/video/jzlcd.c
index 4f729f9..172d5b2 100644
--- a/drivers/video/jzlcd.c
+++ b/drivers/video/jzlcd.c
@@ -22,7 +22,6 @@
 #include <linux/dma-mapping.h>
 #include <linux/platform_device.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <linux/kthread.h>
 
 #include <asm/irq.h>
@@ -1428,11 +1427,11 @@ static int jzlcd_pm_callback(struct pm_dev *pm_dev, pm_request_t req, void *data
 	if (!cfb) return -EINVAL;
 
 	switch (req) {
-	case PM_SUSPEND:
+	case PM_EVENT_SUSPEND:
 		ret = jzfb_suspend();
 		break;
 
-	case PM_RESUME:
+	case PM_EVENT_RESUME:
 		ret = jzfb_resume();
 		break;
 
diff --git a/drivers/video/jzlcd.h b/drivers/video/jzlcd.h
index 45def90..3414c5d 100644
--- a/drivers/video/jzlcd.h
+++ b/drivers/video/jzlcd.h
@@ -664,7 +664,6 @@ do { \
 
 #if defined(CONFIG_SOC_JZ4740)
 #if defined(CONFIG_JZ4740_PAVO) || defined(CONFIG_JZ4740_LYRA) || defined(CONFIG_JZ4740_QI_LB60)
-#define GPIO_PWM	123 /* GP_D27 */
 #define PWM_CHN		4 /* pwm channel */
 #define PWM_FULL	101
 /* 100 level: 0,1,...,100 */
diff --git a/sound/oss/jz_i2s.c b/sound/oss/jz_i2s.c
index 3ed6d3f..f34b50e 100644
--- a/sound/oss/jz_i2s.c
+++ b/sound/oss/jz_i2s.c
@@ -19,7 +19,6 @@
 #include <linux/init.h>
 #include <linux/interrupt.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <sound/driver.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
diff --git a/sound/oss/jz_i2s_4750.c b/sound/oss/jz_i2s_4750.c
index 555d9fb..9dba0cc 100644
--- a/sound/oss/jz_i2s_4750.c
+++ b/sound/oss/jz_i2s_4750.c
@@ -19,7 +19,6 @@
 #include <linux/init.h>
 #include <linux/interrupt.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <sound/driver.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
diff --git a/sound/oss/jz_i2s_dlv_dma_test.c b/sound/oss/jz_i2s_dlv_dma_test.c
index 41c10bd..e9a107a 100644
--- a/sound/oss/jz_i2s_dlv_dma_test.c
+++ b/sound/oss/jz_i2s_dlv_dma_test.c
@@ -19,7 +19,6 @@
 #include <linux/init.h>
 #include <linux/interrupt.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <sound/driver.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
diff --git a/sound/oss/jz_i2s_for_4750.c b/sound/oss/jz_i2s_for_4750.c
index b58ae4d..5465299 100644
--- a/sound/oss/jz_i2s_for_4750.c
+++ b/sound/oss/jz_i2s_for_4750.c
@@ -19,7 +19,6 @@
 #include <linux/init.h>
 #include <linux/interrupt.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <sound/driver.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
diff --git a/sound/oss/jz_pcm_tlv320aic1106_dma.c b/sound/oss/jz_pcm_tlv320aic1106_dma.c
index 2255e08..34d4887 100644
--- a/sound/oss/jz_pcm_tlv320aic1106_dma.c
+++ b/sound/oss/jz_pcm_tlv320aic1106_dma.c
@@ -17,7 +17,6 @@
 #include <linux/init.h>
 #include <linux/interrupt.h>
 #include <linux/pm.h>
-#include <linux/pm_legacy.h>
 #include <sound/driver.h>
 #include <linux/sched.h>
 #include <linux/delay.h>
