From c813c2fcb76ffae92f2728d11f44169a3c3d6834 Mon Sep 17 00:00:00 2001
From: Xiangfu Liu <xiangfu.z@gmail.com>
Date: Sun, 26 Jul 2009 13:00:34 +0800
Subject: [PATCH] fix-compile-jz-nand-error.patch

Signed-off-by: Xiangfu Liu <xiangfu.z@gmail.com>
---
 include/linux/mtd/partitions.h |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/linux/mtd/partitions.h b/include/linux/mtd/partitions.h
index c92b4d4..6090f0e 100644
--- a/include/linux/mtd/partitions.h
+++ b/include/linux/mtd/partitions.h
@@ -38,6 +38,9 @@ struct mtd_partition {
 	char *name;			/* identifier string */
 	u_int32_t size;			/* partition size */
 	u_int32_t offset;		/* offset within the master MTD space */
+	char use_planes;                /* flag to specify whether multiple planes of
+					 * NAND is used in the partition
+					 */
 	u_int32_t mask_flags;		/* master MTD flags to mask out for this partition */
 	struct nand_ecclayout *ecclayout;	/* out of band layout for this partition (NAND only)*/
 	struct mtd_info **mtdp;		/* pointer to store the MTD object */
-- 
1.6.0.4

