Index: b/include/linux/linux_logo.h
===================================================================
--- a/include/linux/linux_logo.h	2010-04-06 15:48:59.816267163 +0800
+++ b/include/linux/linux_logo.h	2010-04-06 15:50:03.743781840 +0800
@@ -48,6 +48,7 @@
 extern const struct linux_logo logo_m32r_clut224;
 extern const struct linux_logo logo_spe_clut224;
 extern const struct linux_logo logo_openwrt_clut224;
+extern const struct linux_logo logo_sakc_clut224;
 
 extern const struct linux_logo *fb_find_logo(int depth);
 #ifdef CONFIG_FB_LOGO_EXTRA
Index: b/drivers/video/logo/Kconfig
===================================================================
--- a/drivers/video/logo/Kconfig	2010-04-06 15:48:03.193766765 +0800
+++ b/drivers/video/logo/Kconfig	2010-04-06 15:49:48.556266652 +0800
@@ -86,4 +86,9 @@
 	bool "224-color OpenWrt Linux logo"
 	default y
 
+config LOGO_SAKC_CLUT224
+	bool "224-color SAKC Linux logo"
+	default y
+
+
 endif # LOGO
Index: b/drivers/video/logo/Makefile
===================================================================
--- a/drivers/video/logo/Makefile	2010-04-06 15:48:18.943783011 +0800
+++ b/drivers/video/logo/Makefile	2010-04-06 15:49:48.556266652 +0800
@@ -16,7 +16,7 @@
 obj-$(CONFIG_LOGO_SUPERH_CLUT224)	+= logo_superh_clut224.o
 obj-$(CONFIG_LOGO_M32R_CLUT224)		+= logo_m32r_clut224.o
 obj-$(CONFIG_LOGO_OPENWRT_CLUT224)	+= logo_openwrt_clut224.o
-
+obj-$(CONFIG_LOGO_SAKC_CLUT224)         += logo_sakc_clut224.o
 obj-$(CONFIG_SPU_BASE)			+= logo_spe_clut224.o
 
 # How to generate logo's
Index: b/drivers/video/logo/logo.c
===================================================================
--- a/drivers/video/logo/logo.c	2010-04-06 15:48:22.896268488 +0800
+++ b/drivers/video/logo/logo.c	2010-04-06 15:49:48.556266652 +0800
@@ -104,6 +104,10 @@
 		/* OpenWrt logo */
 		logo = &logo_openwrt_clut224;
 #endif
+#ifdef CONFIG_LOGO_SAKC_CLUT224
+		/* SAKC logo */
+		logo = &logo_sakc_clut224;
+#endif
 	}
 	return logo;
 }
