diff --git a/board/qi_lb60/qi_lb60.c b/board/qi_lb60/qi_lb60.c
index c23534d..16e8688 100644
--- a/board/qi_lb60/qi_lb60.c
+++ b/board/qi_lb60/qi_lb60.c
@@ -45,47 +45,45 @@ static void gpio_init(void)
 	__gpio_as_i2c();
 
 	/*
+	 * Initialize MSC pins
+	 */
+	__gpio_as_msc();
+	/* __gpio_set_pin(2 + 3 * 32); */
+
+	/*
 	 * Initialize Other pins
 	 */
 /*	unsigned int i;
-	__gpio_as_output( GPIO_AUDIO_POP );
-	__gpio_set_pin( GPIO_AUDIO_POP );
-
-
-	for (i = 0; i < 8; i++){
-		__gpio_as_output( GPIO_KEYOUT_BASE + i );
-		__gpio_set_pin( GPIO_KEYOUT_BASE + i );
+	for (i = 0; i < 8; i++) {
+		__gpio_as_output(GPIO_KEYOUT_BASE + i);
+		__gpio_set_pin(GPIO_KEYOUT_BASE + i);
 	}
 
 	for (i = 0; i < 7; i++){
-		__gpio_as_input( GPIO_KEYIN_BASE + i );
-		__gpio_enable_pull( GPIO_KEYIN_BASE + i );
+		__gpio_as_input(GPIO_KEYIN_BASE + i);
+		__gpio_enable_pull(GPIO_KEYIN_BASE + i);
 	}
-
 	__gpio_as_input( GPIO_KEYIN_8 );
 	__gpio_enable_pull( GPIO_KEYIN_8 );
-
 */
-	__gpio_as_output( GPIO_LCD_CS );
-	__gpio_clear_pin( GPIO_LCD_CS );
+	__gpio_as_output(GPIO_AUDIO_POP);
+	__gpio_set_pin(GPIO_AUDIO_POP);
 
-	__gpio_as_output( GPIO_AMP_EN );
-	__gpio_clear_pin( GPIO_AMP_EN );
+	__gpio_as_output(GPIO_LCD_CS);
+	__gpio_clear_pin(GPIO_LCD_CS);
 
-	__gpio_as_output( GPIO_SDPW_EN );
-	__gpio_set_pin( GPIO_SDPW_EN );
+	__gpio_as_output(GPIO_AMP_EN);
+	__gpio_clear_pin(GPIO_AMP_EN);
 
-	__gpio_as_input( GPIO_SD_DETECT );
-	__gpio_enable_pull( GPIO_SD_DETECT );
+	__gpio_as_output(GPIO_SDPW_EN);
+	__gpio_disable_pull(GPIO_SDPW_EN);
+	__gpio_clear_pin(GPIO_SDPW_EN);
 
-	__gpio_as_input( GPIO_USB_DETECT );
-	__gpio_enable_pull( GPIO_USB_DETECT );
+	__gpio_as_input(GPIO_SD_DETECT);
+	__gpio_disable_pull(GPIO_SD_DETECT);
 
-	/*
-	 * Initialize MSC pins
-	 */
-	__gpio_as_msc();
-	__gpio_set_pin(2 + 3 * 32);
+	__gpio_as_input(GPIO_USB_DETECT);
+	__gpio_enable_pull(GPIO_USB_DETECT);
 }
 
 void board_early_init(void)
diff --git a/cpu/mips/jz_mmc.c b/cpu/mips/jz_mmc.c
index 8a7c310..ec0a518 100644
--- a/cpu/mips/jz_mmc.c
+++ b/cpu/mips/jz_mmc.c
@@ -61,7 +61,7 @@ do {						\
 ({						\
 	int detected = 1;			\
 	__gpio_as_input(GPIO_SD_CD_N);		\
-	if (__gpio_get_pin(GPIO_SD_CD_N))	\
+	if (!__gpio_get_pin(GPIO_SD_CD_N))	\
 		detected = 0;			\
 	detected;				\
 })
