diff --git a/board/qi_lb60/qi_lb60.c b/board/qi_lb60/qi_lb60.c
index d13f2ab..86c33ba 100644
--- a/board/qi_lb60/qi_lb60.c
+++ b/board/qi_lb60/qi_lb60.c
@@ -25,9 +25,11 @@ static void gpio_init(void)
 	__gpio_as_sdram_32bit();
 
 	/*
-	 * Initialize UART0 pins
+	 * Initialize UART0 pins, in Ben NanoNote uart0 and keyin8 use the
+	 * same gpio, init the gpio as uart0 cause a keyboard bug. so for
+	 * end user we disable the uart0
 	 */
-	__gpio_as_uart0();
+	/* __gpio_as_uart0(); */
 
 	/*
 	 * Initialize LCD pins
@@ -52,8 +54,8 @@ static void gpio_init(void)
 		__gpio_as_input(GPIO_KEYIN_BASE + i);
 		__gpio_enable_pull(GPIO_KEYIN_BASE + i);
 	}
-	/* __gpio_as_input( GPIO_KEYIN_8 ); */
-	/* __gpio_enable_pull( GPIO_KEYIN_8 ); */
+	__gpio_as_input(GPIO_KEYIN_8);
+	__gpio_enable_pull(GPIO_KEYIN_8);
 
 	__gpio_as_output(GPIO_AUDIO_POP);
 	__gpio_set_pin(GPIO_AUDIO_POP);
