From c3231a3e316c05b63b573b9d8e6b92970de6037c Mon Sep 17 00:00:00 2001
From: xiangfu <xiangfu.z@gmail.com>
Date: Tue, 21 Apr 2009 22:19:10 +0800
Subject: [PATCH] fix timer callback

---
 arch/mips/jz4740/board-pi.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/mips/jz4740/board-pi.c b/arch/mips/jz4740/board-pi.c
index 8c8544f..309ed9f 100644
--- a/arch/mips/jz4740/board-pi.c
+++ b/arch/mips/jz4740/board-pi.c
@@ -110,5 +110,5 @@ void __init jz_board_setup(void)
 	board_cpm_setup();
 	board_gpio_setup();
 
-	jz_timer_callback = pavo_timer_callback;
+	jz_timer_callback = pi_timer_callback;
 }
-- 
1.6.0.4

