From 6bd871f2f9be7d93524ed031b30af65b8f7ccd61 Mon Sep 17 00:00:00 2001
From: xiangfu <xiangfu.z@gmail.com>
Date: Thu, 23 Apr 2009 13:29:16 +0800
Subject: [PATCH] format the build shell script

---
 build |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/build b/build
index 7c6180b..3051f47 100755
--- a/build
+++ b/build
@@ -25,16 +25,16 @@ fi
 mkdir -p $1
 
 if [ -z "$CROSS_COMPILE" ]; then
-    export CROSS_COMPILE=/opt/mipseltools-gcc412-glibc261/bin/mipsel-linux-
+	export CROSS_COMPILE=/opt/mipseltools-gcc412-glibc261/bin/mipsel-linux-
 fi
 make O=$1 pi_defconfig
 
 
 VERSION=
 if [ -d .git ] ; then
- HEAD=`git show --pretty=oneline | head -n1 | cut -d' ' -f1 | cut -b1-16`
- BRANCH=`git branch | grep ^\* | cut -d' ' -f2 | sed s/-hist//g`
- VERSION=-$PRODUCT\_$BRANCH
+	HEAD=`git show --pretty=oneline | head -n1 | cut -d' ' -f1 | cut -b1-16`
+	BRANCH=`git branch | grep ^\* | cut -d' ' -f2 | sed s/-hist//g`
+	VERSION=-$PRODUCT\_$BRANCH
 fi
 
 if make -j$PARALLEL O=$1 uImage ; then
@@ -45,4 +45,3 @@ fi
 
 exit 1
 
-
-- 
1.6.0.4

