From 345541b72b832c8b78a32ae08a552b4d778e3550 Mon Sep 17 00:00:00 2001
From: xiangfu <xiangfu.z@gmail.com>
Date: Fri, 24 Apr 2009 11:28:38 +0800
Subject: [PATCH] test email

---
 build |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/build b/build
index 3051f47..70d5fcf 100755
--- a/build
+++ b/build
@@ -5,12 +5,9 @@
 #
 # Licensed under GPLv3 or later
 #
-#
 # you need to run this from the top level source dir, but it creates all
 # object files into a subdir given in the first argument, eg
 #
-# 
-#
 # this radically speeds up swapping between build contexts.  Note the config
 # for each build lives in the subdir.
 
-- 
1.6.0.4

