From b70f44257dc5d9560eb8fec6ad48a2a894ef044a Mon Sep 17 00:00:00 2001
From: xiangfu <xiangfu.z@gmail.com>
Date: Wed, 1 Jul 2009 11:29:24 +0800
Subject: [PATCH] fix-make-zImage-erro.path

---
 Makefile           |    2 +-
 arch/mips/Makefile |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 9f73a2b..38800ef 100644
--- a/Makefile
+++ b/Makefile
@@ -191,7 +191,7 @@ SUBARCH := $(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc64/ \
 # Note: Some architectures assign CROSS_COMPILE in their arch/*/Makefile
 
 ARCH		?= mips
-CROSS_COMPILE	?= mipsel-linux-
+CROSS_COMPILE	?= mipsel-openwrt-linux-
 
 # Architecture as present in compile.h
 UTS_MACHINE 	:= $(ARCH)
diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index acad988..ce7dc4a 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -170,7 +170,7 @@ load-$(CONFIG_SOC_JZ4730)	+= 0xffffffff80010000
 #
 
 core-$(CONFIG_SOC_JZ4740)	+= arch/mips/jz4740/
-cflags-$(CONFIG_SOC_JZ4740)	+= -Iinclude/asm-mips/mach-jz4740
+cflags-$(CONFIG_SOC_JZ4740)	+= -Iinclude/asm-mips/mach-jz4740  -Iinclude
 load-$(CONFIG_SOC_JZ4740)	+= 0xffffffff80010000
 
 #
-- 
1.6.0.4

