--- a/drivers/i2c/chips/Kconfig	2009-10-05 13:18:07.000000000 +0200
+++ b/drivers/i2c/chips/Kconfig	2009-12-06 19:10:10.000000000 +0100
@@ -74,4 +74,13 @@
 	  This driver can also be built as a module.  If so, the module
 	  will be called tsl2550.
 
+config N526_LPC
+	tristate "N526 LPC934 coprocessor"
+	depends on JZ4740_N526
+	help
+	  If you say yes here you get support for the N526s NXP LPC934 coprocessor.
+	  It is used as a keyboard controllor and for power management.
+
+	  If you have a N526 you probably want to say Y here.
+
 endmenu
--- a/drivers/i2c/chips/Makefile	2009-10-05 13:18:07.000000000 +0200
+++ b/drivers/i2c/chips/Makefile	2009-12-06 19:03:47.000000000 +0100
@@ -15,6 +15,7 @@
 obj-$(CONFIG_SENSORS_PCF8574)	+= pcf8574.o
 obj-$(CONFIG_PCF8575)		+= pcf8575.o
 obj-$(CONFIG_SENSORS_TSL2550)	+= tsl2550.o
+obj-$(CONFIG_N526_LPC)		+= n526-lpc.o
 
 ifeq ($(CONFIG_I2C_DEBUG_CHIP),y)
 EXTRA_CFLAGS += -DDEBUG
